agent/src/config/mandatoryconfigs.cpp

```diff
@@ -0,0 +1,46 @@
+
+/*
+ * Copyright (C) 2016 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include "mandatoryconfigs.h"
+
+
+MandatoryConfigs::MandatoryConfigs():
+    m_configs(std::vector<std::tuple<std::string, std::string, ConfigType>>()),
+    m_groups(std::vector<std::string>())
+{
+}
+
+void MandatoryConfigs::addConfig(const std::string &group, const std::string &key, ConfigType type)
+{
+    log_debug() << "Adding mandatory config \"" << key << "\" with group \"" << group << "\"";
+    m_groups.push_back(group);
```

Comment:
This means that a mandatory key makes the group it is in mandatory as well, right? That should probably be documented at some stage.